modules/check_in/app/controllers/check_in/v2/appointments_controller.rb

```diff
@@ -36,10 +35,14 @@ def appt_struct_data
         struct.data
       end
 
-      def merge_facilities(appointments)
+      def merge_facilities_and_clinic(appointments)
         appointments.each do |appt|
           unless appt[:locationId].nil?
             appt[:facility] = facility_service.get_facility_with_cache(facility_id: appt[:locationId])
+            unless appt[:clinic].nil?
+              appt[:clinicInfo] =
+                facility_service.get_clinic_with_cache(facility_id: appt[:locationId], clinic_id: appt[:clinic])
+            end
           end
         end
```

Comment:
suggestion for readability:
```suggestion
        appointments.each do |appt|
          next if appt[:locationId].blank?

          appt[:facility] = facility_service.get_facility_with_cache(facility_id: appt[:locationId])

          if appt[:clinic].present?
            appt[:clinicInfo] =
              facility_service.get_clinic_with_cache(facility_id: appt[:locationId], clinic_id: appt[:clinic])
          end
        end
```